### PR TITLE
convert class const's to enum's

### PR DESCRIPTION
**Pull request by ximex**
@MyIgel are such changes wanted?
if yes i would also convert:
- `Handler::(ENV_PRODUCTION|ENV_DEVELOPMENT)`
- `ShiftsFilter::(FILLED_FREE|FILLED_FILLED)`
- `ShiftSignupStatus::(FREE|...|NOT_ARRIVED)`

before i do `ShiftSignupStatus` i wait for the merge of https://github.com/engelsystem/engelsystem/pull/1049 to prevent merge conflicts

### CODE REVIEW

**Comment by ximex** [comment]
The problem is there is already a `Migration` and `Migrate` class. maybe only `Direction`? I extracted the enum for now in an own file

**Comment by ximex** [comment]
rename `MigrateDirection` to `Direction`
also convert `Handler::(ENV_PRODUCTION|ENV_DEVELOPMENT)`

**Comment by MyIgel** [comment]
The Environment as a state is not exception specific and thus could imho eventually be moved to `Engelsystem\Environment`. The Shifts and SignupStatus state would eventually belong to the model but that does not feel right -> to be discussed (@weeman1337 do you have an idea about that?).

**Comment by ximex** [comment]
moved to `Engelsystem\Environment`

File: includes/view/Shifts_view.php [hunk 1]
```diff
@@ -152,12 +153,13 @@ function Shift_view($shift, ShiftType $shifttype, Room $room, $angeltypes_source
...
-    }
-
-    if ($shift_signup_state->getState() == ShiftSignupState::SIGNED_UP) {
-        $content[] = info(__('You are signed up for this shift.'), true);
+    $info = match ($shift_signup_state->getState()) {
+        ShiftSignupStatus::COLLIDES  => info(__('This shift collides with one of your shifts.'), true),
+        ShiftSignupStatus::SIGNED_UP => info(__('You are signed up for this shift.'), true),
+        default => null,
+    };
+    if (!is_null($info)) {
```
Comment (weeman1337) [inline includes/view/Shifts_view.php:161]: Does this really makes the code more understandable? 

Before: Two easily understandable if conditions :white_check_mark: 
Now: a match, default case with null, if not null condition afterwards with assignment

`match` may be great in some cases. But here in my opinion it makes things more complicated (+ adds two lines :D ).

File: includes/model/ShiftSignupState.php [hunk 2]
```diff
@@ -82,18 +37,18 @@ public function combineWith(ShiftSignupState $shiftSignupState)
...
-            ShiftSignupState::FREE      => 80,
-            ShiftSignupState::ANGELTYPE, ShiftSignupState::COLLIDES => 70,
-            ShiftSignupState::OCCUPIED, ShiftSignupState::ADMIN     => 60,
-            default => 0,
+            ShiftSignupStatus::NOT_ARRIVED, ShiftSignupStatus::NOT_YET, ShiftSignupStatus::SHIFT_ENDED => 100,
+            ShiftSignupStatus::SIGNED_UP => 90,
+            ShiftSignupStatus::FREE      => 80,
+            ShiftSignupStatus::ANGELTYPE, ShiftSignupStatus::COLLIDES => 70,
+            ShiftSignupStatus::OCCUPIED,  ShiftSignupStatus::ADMIN    => 60,
+            default => 0, // TODO: shouldn't happen
```
Comment (weeman1337) [inline includes/model/ShiftSignupState.php:51]: TODO or not TODO… :thinking: :wink:

File: src/Database/Migration/Migrate.php [hunk 3]
```diff
@@ -35,15 +29,14 @@ public function __construct(protected SchemaBuilder $schema, protected Applicati
...
-     * @param string $type (up|down)
      */
     public function run(
         string $path,
-        string $type = self::UP,
+        Direction $direction = Direction::UP,
         bool $oneStep = false,
         bool $forceMigration = false
-    ): void {
+    ): void
```
Comment (weeman1337) [inline src/Database/Migration/Migrate.php:38]: Can you run the code style checkers on this PR?

**Review by weeman1337 — COMMENTED** [review]
Looks good in general :+1: 

Minor comments / discussion points.

**Comment by weeman1337** [comment]
> The Shifts and SignupStatus state would eventually belong to the model but that does not feel right -> to be discussed (@weeman1337 do you have an idea about that?).

+1 this can go [the new models](https://github.com/engelsystem/engelsystem/tree/main/src/Models)

File: includes/view/Shifts_view.php (hunk 1 above)
Comment (ximex) [reply includes/view/Shifts_view.php:161]: we could also do this
```php
match ($shift_signup_state->getState()) {
    ShiftSignupStatus::COLLIDES  => $content[] = info(__('This shift collides with one of your shifts.'), true),
    ShiftSignupStatus::SIGNED_UP => $content[] = info(__('You are signed up for this shift.'), true),
    default => null, // do nothing
};
```

**Comment by ximex** [comment]
should i also add this changes?
```patch
diff --git a/includes/controller/users_controller.php b/includes/controller/users_controller.php
index 7037105c..84a41075 100644
--- a/includes/controller/users_controller.php
+++ b/includes/controller/users_controller.php
@@ -5,6 +5,7 @@ use Engelsystem\Models\User\State;
 use Engelsystem\Models\User\User;
 use Engelsystem\ShiftCalendarRenderer;
 use Engelsystem\ShiftsFilter;
+use Engelsystem\ShiftsFilterFilledStatus;
 use Illuminate\Database\Eloquent\Collection;
 use Illuminate\Support\Str;
 
@@ -382,8 +383,8 @@ function shiftCalendarRendererByShiftFilter(ShiftsFilter $shiftsFilter)
     unset($shift_entries_source);
 
     if (
-        in_array(ShiftsFilter::FILLED_FREE, $shiftsFilter->getFilled())
-        && in_array(ShiftsFilter::FILLED_FILLED, $shiftsFilter->getFilled())
+        in_array(ShiftsFilterFilledStatus::FREE->value, $shiftsFilter->getFilled())
+        && in_array(ShiftsFilterFilledStatus::FILLED->value, $shiftsFilter->getFilled())
     ) {
         return new ShiftCalendarRenderer($shifts, $needed_angeltypes, $shift_entries, $shiftsFilter);
     }
@@ -395,7 +396,7 @@ function shiftCalendarRendererByShiftFilter(ShiftsFilter $shiftsFilter)
             $taken = 0;
 
             if (
-                !in_array(ShiftsFilter::FILLED_FILLED, $shiftsFilter->getFilled())
+                !in_array(ShiftsFilterFilledStatus::FILLED->value, $shiftsFilter->getFilled())
                 && !in_array($needed_angeltype['angel_type_id'], $shiftsFilter->getTypes())
             ) {
                 continue;
@@ -414,14 +415,14 @@ function shiftCalendarRendererByShiftFilter(ShiftsFilter $shiftsFilter)
         }
 
         if (
-            in_array(ShiftsFilter::FILLED_FREE, $shiftsFilter->getFilled())
+            in_array(ShiftsFilterFilledStatus::FREE->value, $shiftsFilter->getFilled())
             && $needed_angels_count > 0
         ) {
             $filtered_shifts[] = $shift;
         }
 
         if (
-            in_array(ShiftsFilter::FILLED_FILLED, $shiftsFilter->getFilled())
+            in_array(ShiftsFilterFilledStatus::FILLED->value, $shiftsFilter->getFilled())
             && $needed_angels_count == 0
         ) {
             $filtered_shifts[] = $shift;
diff --git a/includes/model/ShiftsFilter.php b/includes/model/ShiftsFilter.php
index 3eddb627..9000f876 100644
--- a/includes/model/ShiftsFilter.php
+++ b/includes/model/ShiftsFilter.php
@@ -11,16 +11,6 @@ use Engelsystem\Helpers\Carbon;
  */
 class ShiftsFilter
 {
-    /**
-     * Shift is completely full.
-     */
-    public const FILLED_FILLED = 1;
-
-    /**
-     * Shift has some free slots.
-     */
-    public const FILLED_FREE = 0;
-
     /**
      * Has the user "user shifts admin" privilege?
      *
@@ -52,11 +42,11 @@ class ShiftsFilter
         $this->types = $angelTypes;
 
         $this->filled = [
-            ShiftsFilter::FILLED_FREE
+            ShiftsFilterFilledStatus::FREE->value
         ];
 
         if ($user_shifts_admin) {
-            $this->filled[] = ShiftsFilter::FILLED_FILLED;
+            $this->filled[] = ShiftsFilterFilledStatus::FILLED->value;
         }
     }
```

File: src/Database/Migration/Migrate.php (hunk 3 above)
Comment (ximex) [reply src/Database/Migration/Migrate.php]: fixed

File: includes/view/Shifts_view.php (hunk 1 above)
Comment (ximex) [reply includes/view/Shifts_view.php:161]: @weeman1337 suggestion?

File: includes/model/ShiftSignupState.php (hunk 2 above)
Comment (ximex) [reply includes/model/ShiftSignupState.php:51]: should i remove the line? the question is why was it there before?

File: includes/view/ShiftCalendarShiftRenderer.php [hunk 4]
```diff
@@ -171,63 +171,43 @@ private function renderShiftNeededAngeltype(Shift $shift, $shift_entries, $angel
...
+                . button(
+                    page_link_to('user_angeltypes', ['action' => 'add', 'angeltype_id' => $angeltype->id]),
+                    sprintf(__('Become %s'), $angeltype->name),
+                    'btn-sm'
+                ),
+            // Shift collides or user is already signed up: No signup allowed
+            ShiftSignupStatus::COLLIDES, ShiftSignupStatus::SIGNED_UP => $inner_text,
+            // Shift is full
+            ShiftSignupStatus::OCCUPIED => null,
+            default => null, // TODO: shouldn't happen
```
Comment (ximex) [inline includes/view/ShiftCalendarShiftRenderer.php:207]: should i remove the line?

**Comment by ximex** [comment]
> +1 this can go [the new models](https://github.com/engelsystem/engelsystem/tree/main/src/Models)

moved the `ShiftSignupStatus` to `src/Models/Shifts/ShiftSignupStatus`

File: includes/view/Shifts_view.php (hunk 1 above)
Comment (MyIgel) [reply includes/view/Shifts_view.php:161]: Eventually just leaving it as is would be the easiest way here

File: includes/model/ShiftSignupState.php (hunk 2 above)
Comment (MyIgel) [reply includes/model/ShiftSignupState.php:53]: Having 0 here would imho eventually be usefull when extending the enum so keeping it (and removing the ToDo) would be enough.

File: includes/view/ShiftCalendarShiftRenderer.php (hunk 4 above)
Comment (MyIgel) [reply includes/view/ShiftCalendarShiftRenderer.php:208]: Also here: i would keep the defaults

**Comment by ximex** [comment]
@MyIgel change requests done

**Review by MyIgel — CHANGES_REQUESTED** [review]
Please run `phpcs` and all other check tools to ensure the code is up to our code standards.

It is recommended to use [pre-commit-hooks](https://github.com/engelsystem/engelsystem/blob/main/DEVELOPMENT.md#pre-commit-hooks) for that.

File: includes/view/ShiftCalendarShiftRenderer.php [hunk 5]
```diff
@@ -173,63 +174,43 @@ private function renderShiftNeededAngeltype(Shift $shift, $shift_entries, $angel
...
+                ),
+            // No link and add a text hint, when the shift ended
+            ShiftSignupStatus::SHIFT_ENDED => $inner_text . ' (' . __('ended') . ')',
+            // No link and add a text hint, when the shift ended
+            ShiftSignupStatus::NOT_ARRIVED => $inner_text . ' (' . __('please arrive for signup') . ')',
+            ShiftSignupStatus::NOT_YET => $inner_text . ' (' . __('not yet') . ')',
+            ShiftSignupStatus::ANGELTYPE => $angeltype->restricted ?
+                // User has to be confirmed on the angeltype first
+                $inner_text . icon('mortarboard-fill')
+                :
```
Comment (MyIgel) [inline includes/view/ShiftCalendarShiftRenderer.php:198]: I would prefer having the `?` and `:` at the start of the statements instead in a new line to make it clearer.

**Pull request merged**